src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs

```diff
@@ -68,61 +72,95 @@ public class OutputConsoleLogger : INuGetUILogger, IDisposable
                 _solutionEvents = _dte.Events.SolutionEvents;
                 _solutionEvents.AfterClosing += () => { ErrorListTableDataSource.Value.ClearNuGetEntries(); };
                 OutputConsole = await consoleProvider.CreatePackageManagerConsoleAsync();
-            });
+            }
+            ).FileAndForget(TelemetryUtility.CreateFileAndForgetEventName(nameof(OutputConsoleLogger), nameof(OutputConsoleLogger)));
         }
 
         public void Dispose()
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
Interesting technique to getting your AsyncPackage to block on any trailing semaphore users. :)
Since you dispose of the semaphore after exiting the semaphore, it _could_ be that you're disposing it while someone else is using it. Are you sure no one will try to enter it after `OutputConsoleLogger.Dispose` is called?

Reply:
This is MEF shared component. I am guessing it is unlikely that someone will log after disposal. Also, I am assuming that _semaphore will throw if accessed after disposal, and all of those are wrapped into .FileAndForget...